crates/driver/src/auction_converter.rs

```diff
@@ -90,9 +90,15 @@ impl AuctionConverting for AuctionConverter {
 
         tracing::info!(?orders, "got {} orders", orders.len());
 
+        let token_pairs: Vec<_> = orders
+            .iter()
+            .filter(|o| !o.is_liquidity_order)
+            .flat_map(|o| TokenPair::new(o.buy_token, o.sell_token))
+            .collect();
```

Comment:
Not a goal of this refactor but we should deduplicate or collect into a set here so we don't do extra work later. Or is that something all liquidity collectors handle internally?

Reply:
Yeah, currently all the liquidity sources extend the passed in list with the base tokens and deduplicate it individually. This could be moved into the `LiquidityCollector` to avoid duplicate deduplications. 😄 
This would be a very simple follow up PR.